src/Symfony/Component/VarDumper/Dumper/VarDumperOptions.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Dumper;
+
+use Traversable;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ *
+ * @template-implements \ArrayAccess<string, mixed>
+ */
+final class VarDumperOptions implements \ArrayAccess, \Countable, \IteratorAggregate
+{
+    public const FORMAT = '_format';
+    public const TRACE = '_trace';
+    public const TRACE_LIMIT = '_trace_limit';
+    public const MAX_ITEMS = '_max_items';
+    public const MIN_DEPTH = '_min_depth';
+    public const MAX_STRING = '_max_string';
+    public const MAX_DEPTH = '_max_depth';
+    public const MAX_ITEMS_PER_DEPTH = '_max_items_per_depth';
+    public const THEME = '_theme';
+
+    private array $options = [
+        self::FORMAT => null,
+        self::TRACE => false,
+        self::TRACE_LIMIT => 0,
+        self::MAX_ITEMS => null,
+        self::MIN_DEPTH => null,
+        self::MAX_STRING => null,
+        self::MAX_DEPTH => null,
+        self::MAX_ITEMS_PER_DEPTH => null,
+        self::THEME => HtmlDumperTheme::Dark,
+    ];
+
+    public function __construct(iterable $options = [])
+    {
+        $this->options = array_replace($this->options, $options);
+    }
+
+    public function format(?string $format): VarDumperOptions
+    {
+        $this->options[self::FORMAT] = $format;
+
+        return $this;
+    }
+
+    public function trace(): VarDumperOptions
+    {
+        $this->options[self::TRACE] = true;
+
+        return $this;
+    }
+
+    public function traceLimit(?int $limit): VarDumperOptions
+    {
+        if (false === $this->options[self::TRACE]) {
+            throw new \InvalidArgumentException('The trace must be enabled with "VarDumperOptions::trace()" before setting a limit.');
```

Comment:
Exceptions are made to prevent incorrect behaviours. We can just allow setting this option, that is ignored if trace is not enabled.

Reply:
Agree, fixed as well!